### PR TITLE
Preferences/About: draw links with a more readable color, improve defs_urls usage

### PR DESCRIPTION
**Pull request by ronso0**
This aims to guarantee readbility of hyperlinks and file links in Preferences and About.
I added some helpers to blend the palette's text color and the link text color. This should give readable links no matter if the user's theme has the link color defined or not.

In the end the color fix was just a very quick fix but I stumbled upon some inconveniences with how links are created.

* added blendColor() to util/color/color.h to belnd two colors 50/50
* added a small utility to util/string.h to simplify link formatting in ui files:
new class `LinkFormatter`, `QString coloredLinkString`
* added presetSupportLinks() to concentrate creation of controller mapping support links
* some tweaks along the way

Note: this moves some tr strings from ui files to cpp, so we may again need to push the tr sources to transifex.

Previously, I could hardly see the controller file links with my (official) dark OS theme. It seems there's no adequate color defined for links and some Qt default is picked (blue):
image

Now, they're drawn with a blend of the palette's text color / link color:
image

### CODE REVIEW

**Comment by ronso0** [comment]
Sure the link color issue is no bug on our side but (considering myself an advanced user) I couldn't fix it with OS appearance settings or Qt5 Settings so I guess this hack is okay.

**Comment by ronso0** [comment]
WIP again. while trying to understand the clazy warnings I decided to move that link formatting utility to new src/util/linkformatter

**Comment by ronso0** [comment]
Okay, done.
The link string builder utility is now a new class in util/string.h Okay @uklotzde ?

File: src/util/string.h
Comment (daschuer) [inline src/util/string.h:36]:
```suggestion
        return QStringLiteral("<a style=\"color:") + color.name() + QStringLiteral(";\" href=\"") + baseUrl +  extUrl +  QStringLiteral("\">") + text + QStringLiteral("</a>");
```
 The maximum boiler plate version but also the fastest.

Comment (ronso0) [reply src/util/string.h:36]: okay.
so `.arg(` makes the previous way slower?

Comment (daschuer) [reply src/util/string.h:36]: Yes, because the string is parsed for %1, than it is split and concatenated with the replacements. But this does not matter here at all.

Comment (ronso0) [reply src/util/string.h]: okay, I implemented the faster method. (fixed the clazy warnings as well)

**Comment by ronso0** [comment]
@Holzhaus Can you check how that looks like with i3wm?

**Comment by ronso0** [comment]
reminder for myself: check QString > QStringLiteral

File: src/util/string.h
Comment (Holzhaus) [inline src/util/string.h:33]: Why is a class? This could be a free function (or a static one). Or are you planning to add new methods here later?

As an alternative, you could pass in the color once in the constructor instead of passing it in each function call.

Comment (ronso0) [reply src/util/string.h:33]: When we add the ability ro [style DlgTrackproperty](https://github.com/mixxxdj/mixxx/blob/f083c4fdec65c1adea31114c7563c3e989f4404f/src/library/dlgtrackinfo.cpp#L37) and other dialogs (like Preferences) a simple linkString() builder could be helpful. Didn't look into that, yet.

> This could be a free function (or a static one).
As an alternative, you could pass in the color once in the constructor

Can you point me some example? Am still a c++ entrepreneur, without basic knowledge appearantly and having hard times understanding standard coding terms:)

Comment (Holzhaus) [reply src/util/string.h:33]: A free function is just a function that does not belong to a class (i.e. without `class Foo {}` around it). E.g.:

```cpp
bool negate(bool value) {
    return !value;
}
```

And you call it like this:

```cpp
bool myval = negate(val);
```

This example only uses free functions: https://www.learn-cpp.org/en/Functions

A static method looks like this:

```cpp
class Foo {
  public: 
    static bool negate(bool value) {
        return !value;
    }
  private:
    // Make constructor private to prevent creating an object of this class. 
    Foo() {}
} 
```

And you call it like this:

```cpp
bool myval = Foo::negate(val);
```

https://www.learncpp.com/cpp-tutorial/static-member-functions/

Comment (ronso0) [reply src/util/string.h:33]: Thanks a lot for the explanation! I will look into that.

Regarding
> When we add the ability ro style DlgTrackproperty and other dialogs (like Preferences) a simple linkString() builder could be helpful. Didn't look into that, yet.

is it okay to merge as it is now?

Comment (ronso0) [reply src/util/string.h]: I learned: a free function in a header (string.h here) it needs to be `inline`
https://github.com/mixxxdj/mixxx/pull/3464/commits/850bbdf347c9b5ec72dd2539c25301f26abf5942

Comment (ronso0) [reply src/util/string.h]: Ready again..

**Review by Holzhaus — APPROVED** [review]
LGTM, thank you!

**Comment by Holzhaus** [comment]
Waiting for CI.

**Comment by ronso0** [comment]
> Waiting for CI.

all greeeen :)

**Comment by ronso0** [comment]
I'm not sure how I made https://github.com/mixxxdj/mixxx/pull/3464/commits/a0e791157980c5fef0ea6fa3eb06b5fc58835a4f slip in here... whoopsy!
gave me some headaches when I merged 2.3 into main locally

**Comment by uklotzde** [comment]
Regression? https://bugs.launchpad.net/mixxx/+bug/1910906

**Comment by ronso0** [comment]
yup, from a first glance it looks like.
but I built before we merged, also CI passed, and res builder would fail if the icons are not found.
also #3535 CI succeeded..

@uklotzde Please give the final Okay for #3484 so i can squash and merge to 2.3, then merge 2.3 into main.

**Pull request merged**
